vision/pipeline.py

```diff
@@ -88,44 +93,92 @@ def run(self, prev_state):
             state = self.flight_communication.get_nowait()
         except Empty:
             state = prev_state
-
         ##
+
         bboxes = []
 
+        flags = {}
+
+        state = "module_detection"
+
         if state == "early_laps":  # navigation around the pylons
-            bboxes = self.obstacle_finder.find(color_image, depth_image)
+            flags = ObstacleDetectionFlags()
+
+            try:
+                bboxes = self.obstacle_finder.find(color_image, depth_image)
+                self.obstacle_tracker.update(bboxes)
+            except:
+                flags.obstacle_finder = False
 
-            obstacle_tracker.update(bboxes)
-            bboxes = obstacle_tracker.getPersistentObstacles()
+            try:
```

Comment:
For the cases where one algorithm is depends on another, you should check whether the previous algorithm crashed before trying to run the dependent algorithm.